pkg/config/resolver.go

```diff
@@ -243,32 +260,57 @@ func (r *Resolver) Queue() (*kubernetes.Queue, error) {
 // RegisterNewResultsListener resolver method
 func (r *Resolver) RegisterNewResultsListener() {
 	targets := r.TargetClients()
-	if targets.Empty() {
-		return
-	}
 
-	newResultListener := listener.NewResultListener(r.SkipExistingOnStartup(), r.ResultCache(), time.Now())
+	newResultListener := listener.NewResultListener(r.ResultCache(), time.Now())
 	r.resultListener = newResultListener
 	r.EventPublisher().RegisterListener(listener.NewResults, newResultListener.Listen)
 
 	r.EventPublisher().RegisterPostListener(listener.CleanUpListener, listener.NewCleanupListener(context.Background(), targets))
 }
 
 // RegisterSendResultListener resolver method
-func (r *Resolver) RegisterSendResultListener() {
-	targets := r.TargetClients()
+func (r *Resolver) RegisterSendResultListener(targetChan chan targetconfig.TcEvent) {
+	registerFunc := func(targets *target.Collection) {
+		// if the client has skip existing preload its cache with the existing results
+		for _, client := range targets.Targets() {
+			if client.Client.SkipExistingOnStartup() {
+				client.Client.SetCache(r.resultCache.Clone())
+			}
+		}
 
-	if targets.Empty() {
-		return
-	}
+		r.resultListener.RegisterListener(listener.NewSendResultListener(targets))
+		r.resultListener.RegisterScopeListener(listener.NewSendScopeResultsListener(targets))
+		r.resultListener.RegisterSyncListener(listener.NewSendSyncResultsListener(targets))
+	}
+
+	go func() {
+		r.logger.Info("starting listener loop")
+		for {
+			select {
+			case event := <-targetChan:
+				// clear existing listeners and create new ones
+				r.logger.Info(fmt.Sprintf("received targetconfig event of type %s", event.Type))
+				r.resultListener.ResetListeners()
+				registerFunc(event.Targets)
+
+				if event.Type == targetconfig.CreateTcEvent && event.RestartPolrInformer {
+					if len(event.Targets.Clients()) < r.targetConfigClient.TargetConfigCount() && !r.targetConfigClient.HasSynced() {
+						continue
+					}
+					r.polrRestartCh <- struct{}{}
```

Comment:
also needs cleanup, I would appreciate if we could just remove everything we do not need right now. You can keep it in a different branch for later but I do not want to role out "dead code" which could have a negative impact.